src/hooks/useOrders.ts

```diff
@@ -1,34 +1,90 @@
 import { useEffect, useCallback, useRef } from 'react'
-// eslint-disable-next-line @typescript-eslint/camelcase
-import { unstable_batchedUpdates } from 'react-dom'
+import { unstable_batchedUpdates as batchedUpdates } from 'react-dom'
 
 import useGlobalState from './useGlobalState'
 import { overwriteOrders, updateOffset, updateOrders } from 'reducers-actions/orders'
 
 import { useWalletConnection } from './useWalletConnection'
 import useSafeState from './useSafeState'
-import { exchangeApi } from 'api'
-import { AuctionElement } from 'api/exchange/ExchangeApi'
+import { exchangeApi, web3 } from 'api'
+import { AuctionElement, PendingTxObj } from 'api/exchange/ExchangeApi'
 import { useCheckWhenTimeRemainingInBatch } from './useTimeRemainingInBatch'
+import { removePendingOrdersAction } from 'reducers-actions/pendingOrders'
+import { REFRESH_WHEN_SECONDS_LEFT, EMPTY_ARRAY } from 'const'
 
 interface Result {
   orders: AuctionElement[]
+  pendingOrders: PendingTxObj[]
   forceOrdersRefresh: () => void
   isLoading: boolean
 }
 
-const REFRESH_WHEN_SECONDS_LEFT = 60 // 1min before batch done
-// solutions submitted at this point
-
 export function useOrders(): Result {
   const { userAddress, networkId, blockNumber } = useWalletConnection()
   const [
     {
       orders: { orders, offset },
+      pendingOrders,
     },
     dispatch,
   ] = useGlobalState()
 
+  // Handle Pending Orders
+  // pending orders are saved in global app state as well as local storage
+  // users can refresh page after creating orders and see them after reload thanks for local storage
+  // uses this effect hook to:
+  //   1. listen to block updates and filter recently mined transactions based on their hash
+  //   2. remove any orders that may have been re-sent at a higher gas price
+  const currentPendingOrders = (userAddress && networkId && pendingOrders[networkId]?.[userAddress]) || EMPTY_ARRAY
+  useEffect(() => {
+    // Don't fire if there are no pending orders...
+    if (userAddress && networkId && currentPendingOrders.length > 0) {
+      const managePendingOrders = async (): Promise<void> => {
+        const latestBlock = await web3.eth.getBlock(blockNumber || 'latest')
+
+        // Set from block transaction array for easy lookup
+        const transactionsSet = new Set(latestBlock.transactions)
```

Comment:
Is it really necesary to check on every block over the whole list of transactions?
It looks a bit of overkilling

Reply:
well, only if there are pending orders i guess :shrug: